container/test-impl-base/src/main/java/org/jboss/arquillian/container/test/impl/client/deployment/DeploymentGenerator.java

```diff
@@ -69,13 +69,19 @@
     private Instance<ProtocolRegistry> protocolRegistry;
 
     public void generateDeployment(@Observes GenerateDeployment event) {
-        DeploymentScenarioGenerator generator = serviceLoader.get().onlyOne(
-            DeploymentScenarioGenerator.class, AnnotationDeploymentScenarioGenerator.class);
+        //DeploymentScenarioGenerator generator = serviceLoader.get().onlyOne(
+        //    DeploymentScenarioGenerator.class, AnnotationDeploymentScenarioGenerator.class);
+
+        final Collection<DeploymentScenarioGenerator> deploymentScenarioGenerators =
+            serviceLoader.get().all(DeploymentScenarioGenerator.class);
 
         DeploymentScenario scenario = new DeploymentScenario();
 
-        for (DeploymentDescription deployment : generator.generate(event.getTestClass())) {
-            scenario.addDeployment(deployment);
+        for (DeploymentScenarioGenerator generator : deploymentScenarioGenerators) {
+
+            for (DeploymentDescription deployment : generator.generate(event.getTestClass())) {
```

Comment:
Makes me wonder if we really have to generate all deployments sequentially here. WDYT @lordofthejars @MatousJobanek?

Reply:
For me it is the safest way and also follows how all Arquillian works which is that almost everything works sequentially.